From 99a9e5c1aaa65e39d71b10b0b5f9a4373ef600ba Mon Sep 17 00:00:00 2001
From: Vinayakumar B <vinayakumarb@apache.org>
Date: Tue, 23 Feb 2016 14:39:24 +0530
Subject: [PATCH 1341/2848] HDFS-8578. On upgrade, Datanode should process all
 storage/data dirs in parallel.  Contributed by
 vinayakumarb and szetszwo Addendum to fix
 compilation

(cherry picked from commit 33b961ee88097d34f6b1d87ca72389613a4eb1dc)

Change-Id: I1237da47990703d4504cab39e591b6b3ee4290b0
---
 .../hadoop/hdfs/server/datanode/DataStorage.java   |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataStorage.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataStorage.java
index a4a5578..810076c 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataStorage.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataStorage.java
@@ -22,6 +22,7 @@
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InterruptedIOException;
 import java.io.RandomAccessFile;
 import java.nio.channels.FileLock;
 import java.nio.channels.OverlappingFileLockException;
@@ -55,7 +56,6 @@
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.fs.permission.FsPermission;
 import org.apache.hadoop.hdfs.DFSConfigKeys;
-import org.apache.hadoop.hdfs.DFSUtilClient;
 import org.apache.hadoop.hdfs.HdfsConfiguration;
 import org.apache.hadoop.hdfs.protocol.Block;
 import org.apache.hadoop.hdfs.protocol.HdfsConstants;
@@ -443,7 +443,8 @@ static int getParallelVolumeLoadThreadsNum(int dataDirs, Configuration conf) {
         } catch (ExecutionException e) {
           LOG.warn("Failed to upgrade storage directory " + t.dataDir, e);
         } catch (InterruptedException e) {
-          throw DFSUtilClient.toInterruptedIOException("Task interrupted", e);
+          throw (InterruptedIOException) new InterruptedIOException(
+              "Task interrupted").initCause(e);
         }
       }
     }
@@ -492,7 +493,8 @@ static int getParallelVolumeLoadThreadsNum(int dataDirs, Configuration conf) {
           LOG.warn("Failed to upgrade storage directory " + t.dataDir
               + " for block pool " + bpid, e);
         } catch (InterruptedException e) {
-          throw DFSUtilClient.toInterruptedIOException("Task interrupted", e);
+          throw (InterruptedIOException) new InterruptedIOException(
+              "Task interrupted").initCause(e);
         }
       }
     }
-- 
1.7.9.5

